docs/source/healthcheck.rst

```diff
@@ -0,0 +1,60 @@
+Healthchecks
+============
+
+If your code blocks for too long in the main thread, the consumer can turn
+unhealthy.
+
+Kafka has a setting called ``max.poll.interval.ms`` for this that tells Kafka
+to kick the consumer out of the broker after this many milliseconds of not polling.
+
+You can pass this option into :py:class:`arroyo.backends.kafka.consumer.KafkaConsumer` like so:
+
+.. code-block:: Python
+
+   consumer = KafkaConsumer(
+       {
+           "max.poll.interval.ms": 5 * 60, # default 5 minutes
+       }
+   )
+
+However, this will not shut down the consumer, it will just keep running doing
+nothing (because it is blocked in the main thread). You want a pod-level
+healthcheck as well.
+
+Arroyo supports touching a file repeatedly from the main thread to indicate health.
+
+.. code-block:: Python
+
+    processor = StreamProcessor(
+        consumer=consumer,
+        # Arroyo will touch/create this file at least every second.
+        healtcheck_file="/tmp/health.txt"
+    )
+
+The Kubernetes `liveness
+<https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/>`_
+command would look like:
+
+.. code-block:: YAML
+
+    apiVersion: v1
+    kind: Pod
+    metadata:
+      labels:
+        test: liveness
+      name: liveness-exec
+    spec:
+      containers:
+      - name: liveness
+        image: registry.k8s.io/busybox
+        args:
+        - /bin/sh
+        - -c
+        - rm /tmp/health.txt
+        livenessProbe:
+          exec:
+            command:
+            - cat
+            - /tmp/healthy
```

Comment:
This k8s example doesn't align with the code. I'm guessing you copy pasted it. This might be a dump question, but in the case of arroyo, would the liveness command would have to be a unix command to checks the mtime of the file?

Reply:
Yeah that's a copypaste error. I accidentally replaced the container entrypoint with the healthcheck command. fixed

I could check mtime, instead I am deleting the file every n seconds. If it stops being recreated, the consumer is dead